realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -637,47 +534,61 @@ protected static synchronized void validateAgainstExistingConfigurations(RealmCo
     /**
      * Deletes the Realm file defined by the given configuration.
      */
-    protected static synchronized boolean deleteRealm(RealmConfiguration configuration) {
-        if (isFileOpen(configuration)) {
-            throw new IllegalStateException("It's not allowed to delete the file associated with an open Realm. " +
-                    "Remember to close() all the instances of the Realm before deleting its file.");
-        }
+    protected static boolean deleteRealm(final RealmConfiguration configuration) {
+        final AtomicBoolean realmDeleted = new AtomicBoolean(true);
+
+        RealmCache.invokeWithGlobalRefCount(configuration, new RealmCache.Callback() {
+            @Override
+            public void onResult(int count) {
```

Comment:
see my previous comments about the callback, you can just call 
```java
int count = RealmCache.invokeWithGlobalRefCount(configuration);
```
and proceed with the rest of the logic